File: examples/tiny-py-app/src/math.py
"""Math utility functions."""


def add(a: int, b: int) -> int:
    return a + b


def multiply(a: int, b: int) -> int:
    return a * b


def compute(op: str, a: int, b: int) -> int:
    if op == "add":
        return add(a, b)
    elif op == "multiply":
        return multiply(a, b)
    raise ValueError(f"Unknown operation: {op}")

File: examples/tiny-ts-app/src/math.ts
export function add(a: number, b: number): number {
  return a + b;
}

export function multiply(a: number, b: number): number {
  return a * b;
}

export type Operation = "add" | "multiply";

export function compute(op: Operation, a: number, b: number): number {
  switch (op) {
    case "add":
      return add(a, b);
    case "multiply":
      return multiply(a, b);
  }
}
